src/utils/IntlWrapper.ts

```diff
@@ -26,139 +28,111 @@ const dateUnitToMs = {
   seconds: 1000,
 } as const;
 
-enum Rules {
-  DurationFormat,
-  Progress,
-  Range,
-  DateTime,
-}
+export class IntlDuration {
+  protected duration?: durationFormat;
 
-export class IntlWrapper {
-  // Output variables
-  protected output: {
-    text: string;
-    progress: {
-      time?: durationFormat;
-      isFuture: boolean;
-    };
-  } = {
-    text: '',
-    progress: {
-      isFuture: false,
-    },
-  };
+  protected relativeTime?: number;
 
-  // Input variables
-  protected from: Date | number = new Date();
+  protected isFuture = false;
 
-  protected to: Date | number = new Date();
+  protected isNow = false;
 
-  protected locale: Intl.LocalesArgument = api.storage.lang('locale');
+  protected locale: Intl.LocalesArgument;
 
-  protected duration?: durationFormat;
+  protected durationFormatStyle: durationFormatStyle;
 
-  protected timestamp?: number;
+  protected durationStyle: durationStyle;
 
   // @ts-expect-error surely it works
   protected isFallback = !Intl.DurationFormat;
 
-  constructor() {
+  constructor(
+    input: number | durationFormat = { minutes: 0 },
+    style: durationStyle = 'Duration',
+    convertFrom?: keyof typeof dateUnitToMs,
+    format: durationFormatStyle = 'narrow',
+    locale: Intl.LocalesArgument = api.storage.lang('locale'),
+  ) {
+    if (typeof input === 'number') this.setRelativeTime(input, convertFrom);
+    if (typeof input === 'object') this.setDuration(input);
+    this.locale = locale;
+    this.durationStyle = style;
+    this.durationFormatStyle = format;
     return this;
   }
 
-  // Setters
-  setTimestamp(timestamp: number, convertFrom?: keyof typeof dateUnitToMs) {
-    this.setDate(
-      convertFrom ? Number(IntlWrapper.toTimestamp(timestamp, convertFrom)) : Number(timestamp),
-    );
+  setLocale(locale: Intl.LocalesArgument) {
+    this.locale = locale;
     return this;
   }
 
-  setDates(from: Date | number | string, to: Date | number | string) {
-    this.from = new Date(from);
-    this.to = new Date(to);
-    this.timestamp = this.from.getTime();
+  setRelativeTime(relativeTime: number, convertFrom?: keyof typeof dateUnitToMs) {
+    this.relativeTime = Number(relativeTime);
+    if (convertFrom) this.toTimestamp(convertFrom);
```

Comment:
So you mean completely working with `durationFormat` object? The only use for `relativeTime` was that the input could be like 150 (mins) and that number easily convertable to timestamp and then to `durationFormat`. If I use only `durationFormat` as an input param I could not convert 150 (in this case `{minutes: 150}`) to `{hours: 2, minutes: 30}` because `timestampToTime` only uses timestamps. I guess I could make another function to convert something like  `{minutes: 150}` to `{hours: 2, minutes: 30}`?